pkg/argocd/argocd_test.go

```diff
@@ -1005,6 +1005,89 @@ func Test_SetHelmImage(t *testing.T) {
 		require.Error(t, err)
 	})
 
+	t.Run("Test set Helm image parameters when two paths are provided", func(t *testing.T) {
+		app := &v1alpha1.Application{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "test-app",
+				Namespace: "testns",
+				Annotations: map[string]string{
+					fmt.Sprintf(common.HelmParamImageNameAnnotation, "foobar"): "foobar.image.name,foobar2.image.name",
+					fmt.Sprintf(common.HelmParamImageTagAnnotation, "foobar"):  "foobar.image.tag, foobar2.image.tag", // Space is expected
+				},
```

Comment:
also need to handle cases like below to exclude empty slice elements caused by extra separators.
`foobar.image.name,foobar2.image.name,`
`foobar.image.name,,,foobar2.image.name`

Reply:
I edited the test to take care about this point and fixed the code according to